### DISCUSSION TITLE
Inexplicable error connecting to a HTTP server

### DISCUSSION THREAD

**Discussion by Coooz** [nextgenhealthcare/connect#6102] [Q&A] [answered]
Hi,

I notice that in a HTTP Sender destination of one of my channels, all messages get queued; the destination queues messages on failure. Debugging the channel reveals this error:
```
HTTP Sender error
 ERROR MESSAGE: Error connecting to HTTP server
 java.lang.NullPointerException
 	at com.mirth.connect.connectors.http.HttpDispatcher.send(HttpDispatcher.java:245)
 	at com.mirth.connect.donkey.server.channel.DestinationConnector.handleSend(DestinationConnector.java:904)
 	at com.mirth.connect.donkey.server.channel.DestinationConnector.run(DestinationConnector.java:730)
 	at java.base/java.lang.Thread.run(Thread.java:834)
```
However, clicking the [Test connection] button in the destination reveals that the connection is successful. If I view the Sent option of a queued message I see this...
```
URL: https://server.fire.ly/Patient/
METHOD: GET

[HEADERS]
Accept: application/fhir+xml

[PARAMETERS]
_id: 20beea94-5cce-4f34-8f68-160412ae0dde

[CONTENT]
```
...which is perfectly alright. GETting the url in Postman returns the requested xml.

How can I establish a connection to the server? I use Mirth Connect 4.3.0.

**Comment by narupley** [comment]
What is actually set in your HTTP Sender destination settings? Are you using variables? My only guess is that maybe a variable (like your URL) doesn't exist at first, but then exists later so it replaced on a subsequent queue attempt or something.

**Comment by Coooz** [comment]
@narupley 
No, I am not using variables. The only thing that seems out of place is the warning that SSL is not configured:
![image](https://github.com/nextgenhealthcare/connect/assets/88899312/726d5116-d4df-4d1d-a275-e82d42cf59b2)
But then, SSL is not configured in Postman either (I don't even have an account there and only use the lightweight client), and Postman works fine - _and_ I would expect the [Test connection] button to come up with a message that the connection failed if I had to do some additional configuration on my part.

I have included the channel. If you would be so kind as to see why it won't work, then that is much appreciated!
[Get patient by id.xml.txt](https://github.com/nextgenhealthcare/connect/files/14309642/Get.patient.by.id.xml.txt)

**Reply by pacmano1** [reply, marked as answer]
You are using a variable for patient ID.   

On the channel remove the queue on failure (just let it error for now)

Also remove the query parameter and just try: `https://server.fire.ly/Patient/${pat_id}` in the URL.

**Reply by pacmano1** [reply]
Lastly send your postman request do a dumb listener that echoes the exact contents back to the console (i.e. not Mirth!).   Do the same with the mirth channel.

If you have node installed you can start a dumb listener with `npx http-echo-server`

You can either override the hosts on the mirth server or just temporarily send it to a machine mirth can connect to.

Compare the posts on the console from the echo server.  It is nearly always clear what is breaking.

**Reply by Coooz** [reply]
@pacmano1 Removing the query parameter and specifying it in the url as you suggested did the trick. Thanks a lot!
